central/role/datastore/defaults.go

```diff
@@ -0,0 +1,220 @@
+			permissions.View(resources.Node),
+			permissions.View(resources.Deployment),
+			permissions.View(resources.Image),
+			permissions.View(resources.WatchedImage),
+			permissions.View(resources.VulnerabilityReports),
+			permissions.View(resources.WorkflowAdministration),
+			permissions.Modify(resources.VulnerabilityManagementRequests),
+		},
+	},
+	accesscontrol.VulnerabilityManagementAdmin: {
+		idSuffix:    "vulnmgmtadmin",
+		postgresID:  accesscontrol.DefaultPermissionSetIDs[accesscontrol.VulnerabilityManagementAdmin],
+		description: "For users: Use it to provide administrative access to analyze vulnerabilities, generate reports, and manage risk acceptance process",
+		resourceWithAccess: []permissions.ResourceWithAccess{
+			permissions.View(resources.Cluster),
+			permissions.View(resources.Node),
+			permissions.View(resources.Namespace),
+			permissions.View(resources.Deployment),
+			permissions.View(resources.Image),
+			permissions.View(resources.Integration),
+			permissions.Modify(resources.WatchedImage),
+			permissions.Modify(resources.VulnerabilityManagementRequests),
+			permissions.Modify(resources.VulnerabilityManagementApprovals),
+			permissions.Modify(resources.VulnerabilityReports),
+			permissions.Modify(resources.WorkflowAdministration),
+		},
+	},
+}
+
+func getDefaultRoles() []*storage.Role {
+	roles := make([]*storage.Role, 0, len(defaultPermissionSets))
+	for _, roleName := range accesscontrol.DefaultRoleNames.AsSlice() {
+		attributes, found := defaultPermissionSets[roleName]
+		if !found {
+			utils.Should(errors.Errorf("Default role %s does not have permission set defined", roleName))
+		}
+
+		// Historically, default permission sets and roles have had same name to make it easier for the user to quickly map one to the other.
+		permissionSet := getDefaultPermissionSet(roleName)
+		role := &storage.Role{
+			Name:          roleName,
+			Description:   attributes.description,
+			AccessScopeId: rolePkg.AccessScopeIncludeAll.GetId(),
+			Traits: &storage.Traits{
+				Origin: storage.Traits_DEFAULT,
+			},
+			PermissionSetId: permissionSet.Id,
+		}
+		roles = append(roles, role)
+	}
+	return roles
+}
+
+func getDefaultPermissionSets() []*storage.PermissionSet {
+	permissionSets := make([]*storage.PermissionSet, 0, len(defaultPermissionSets))
+
+	for name, attributes := range defaultPermissionSets {
+		resourceToAccess := permissionsUtils.FromResourcesWithAccess(attributes.resourceWithAccess...)
+
+		permissionSet := &storage.PermissionSet{
+			Id:               attributes.getID(),
+			Name:             name,
+			Description:      attributes.description,
+			ResourceToAccess: resourceToAccess,
+			Traits: &storage.Traits{
+				Origin: storage.Traits_DEFAULT,
+			},
+		}
+		permissionSets = append(permissionSets, permissionSet)
+	}
+
+	return permissionSets
+}
+
+func getDefaultPermissionSet(name string) *storage.PermissionSet {
+	for pname := range accesscontrol.DefaultPermissionSetIDs {
```

Comment:
No need to iterate over this, can simply use `_, ok := accesscontrol.DefaultPermissionSetIDs[name]`.

Reply:
ah, yes!